pandas/io/common.py

```diff
@@ -631,6 +646,10 @@ def get_handle(
     compression_args = dict(ioargs.compression)
     compression = compression_args.pop("method")
 
+    # GH 24306
+    if mode not in ["r", "rb"] and is_path:  # Only for write methods
```

Comment:
the GH reference is not needed, you can move the comment there instead

Reply:
Could probably be `"r" not in mode and is_path`. Or also check that the parent directory exists when reading.

Reply:
> Or also check that the parent directory exists when reading.

Current error message for reading is along the lines of `FileNotFoundError: File or directory <PATH> does not exist`. So it looks intuitive enough to me.

I think the main point of confusion for beginners is that they are expecting the file to be created for writing methods, so a `FileNotFoundError` doesn't make much sense, like in this SO post https://stackoverflow.com/questions/47143836/pandas-dataframe-to-csv-raising-ioerror-no-such-file-or-directory

Reply:
done